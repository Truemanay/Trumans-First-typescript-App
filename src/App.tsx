/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 */

import React, { FC } from "react";
import { createStackNavigator } from "@react-navigation/stack";
import { NavigationContainer } from "@react-navigation/native";
import { P01, P02 } from "./screens";
import { kRoute } from "./kRoute";

const Stack = createStackNavigator();

const App: FC = () => {
  return (
    <NavigationContainer>
      <Stack.Navigator>
        <Stack.Screen
          name={kRoute.P01}
          component={P01}
          options={{ headerShown: false }}
        />
        <Stack.Screen
          name={kRoute.P02}
          component={P02}
          options={{ headerShown: false }}
        />
      </Stack.Navigator>
    </NavigationContainer>
  );
};

export default App;
