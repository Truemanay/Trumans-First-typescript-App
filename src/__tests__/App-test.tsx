/**
 * @format
 */
import "react-native";
import React from "react";
import { shallow } from "enzyme";
import App from "../App";

// Creating a suite of tests related to running the app
describe("Running the app", () => {
  // Check to see if app renders like it has previously, by snapshotting.
  describe("Rendering the app", () => {
    it("should render correctly", () => {
      // Creates a shallow render of the app.
      const appRendered = shallow(<App />);
      // Expects the snapshot this render makes to match prior snapshot.
      expect(appRendered).toMatchSnapshot();
    });
  });
});
