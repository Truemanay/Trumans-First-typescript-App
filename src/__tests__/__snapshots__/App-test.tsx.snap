// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Running the app Rendering the app should render correctly 1`] = `
<ForwardRef(NavigationContainer)>
  <StackNavigator>
    <Screen
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "compare": null,
          "type": [Function],
        }
      }
      name="P01"
      options={
        Object {
          "headerShown": false,
        }
      }
    />
    <Screen
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "compare": null,
          "type": [Function],
        }
      }
      name="P02"
      options={
        Object {
          "headerShown": false,
        }
      }
    />
    <Screen
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "compare": null,
          "type": [Function],
        }
      }
      name="P03"
      options={
        Object {
          "headerShown": false,
        }
      }
    />
  </StackNavigator>
</ForwardRef(NavigationContainer)>
`;
